src/codegeneration/ArrowFunctionTransformer.js

```diff
@@ -46,6 +59,68 @@ export class ArrowFunctionTransformer extends TempVarTransformer {
    * function body and return statement if needed.
    */
   transformArrowFunctionExpression(tree) {
+    if (this.inDerivedClass_ && this.inConstructor_) {
+      return this.transformUsingCommaExpression_(tree);
+    }
+    return this.transformUsingTempVar_(tree);
+  }
+
+  // This transforms using a comman expression to prevent reading this
+  // before the current location.
+  //
```

Comment:
Maybe:
Transform the arrow function using a comma expression to avoid needing to read 'this' earlier in the function.